File: tools/telemetry/telemetry/util_unittest.py
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
# Copyright (c) 2012 The Chromium Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.
import unittest

from telemetry import util

class TestWait(unittest.TestCase):
  @staticmethod
  def testNonTimeout():
    def test():
      return True
    util.WaitFor(test, 0.1)

  def testTimeout(self):
    def test():
      return False
    self.assertRaises(util.TimeoutException, lambda: util.WaitFor(test, 0.1))

File: dbus/util_unittest.cc
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2014 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "dbus/util.h"
#include "testing/gtest/include/gtest/gtest.h"

namespace dbus {

TEST(UtilTest, GetAbsoluteMemberName) {
  EXPECT_EQ("InterfaceName.MemberName",
            GetAbsoluteMemberName("InterfaceName", "MemberName"));
  EXPECT_EQ(".", GetAbsoluteMemberName("", ""));
}

}  // namespace dbus
